.github/workflows/upload.yml

```
@@ -29,20 +38,19 @@ jobs:
           mv mrmustard src
 
       - name: Install only test dependencies
-        run: uv sync --inexact --no-install-project --all-extras
+        run: uv sync --only-dev --inexact --no-install-project
 
       - name: Run tests
         run: uv run --no-project pytest tests -p no:warnings --tb=native --backend=tensorflow
 
-      - name: Publish
-        uses: pypa/gh-action-pypi-publish@release/v1
-        with:
-          user: __token__
-          password: ${{ secrets.PIPY_TOKEN }}
+      # - name: Publish
+      #   uses: pypa/gh-action-pypi-publish@release/v1
+      #   with:
+      #     user: __token__
+      #     password: ${{ secrets.PIPY_TOKEN }}
 
       - name: Install MrMustard from pypi
-        run: uv pip install mrmustard
+        run: uv pip install --pre --upgrade mrmustard
```

The `--pre` flag will ensure we can get the pre-release version (see https://pip.pypa.io/en/stable/cli/pip_install/#pre-release-versions).

---

although `uv pip` is similar to `pip`, it seems different here. According to its helptext, this should do it:
```suggestion
        run: uv pip install --prerelease=allow --upgrade mrmustard
```

---

on another note, maybe instead of avoiding `uv run`, we just run `uv pip uninstall mrmustard` before installing to make extra-sure? basically I want to know that it's not just using the existing installation, because that's what happened last time.

---

Won't `--prerelease=allow` install pre-release versions for all dependencies? Note that I did test this both locally and at https://github.com/XanaduAI/MrMustard/actions/runs/16473307388/job/46568016618

---

And sure, we could uninstall first as well.

---

Better yet, I could clear the virtual environment?

---

Hmm... `--no-cache-dir` is maybe the best call, actually.